docs/content/users/usage/architecture.md

```diff
@@ -67,6 +67,9 @@ A project’s `.ddev` directory can be intimidating at first, so let’s take a
 `web-build` directory
 : Can be used to provide a [custom Dockerfile](../extend/customizing-images.md) for the web container.
 
+`web-entrypoint.d` directory
+: Custom scripts (named `*.sh`) in this directory will be run during web container startup, before the php-fpm server or other daemons are run. This is an advanced topic, so don't put anything here unless you know why you're doing it. It's useful for introducing environment variables into the context of the nginx and php-fpm servers, for example.
```

Comment:
Restructure to lead with the example, then explain _why_ caution is warranted. “This is an advanced topic” subtly sounds like a status judgment (“this is only the gifted students!”) rather than plainly explaining why someone should be careful.

My reason could be wrong, so that could need to be revised for accuracy. 😅 

```suggestion
: Custom scripts (named `*.sh`) in this directory will be run during web container startup, before the php-fpm server or other daemons are run. This can be useful, for example, for introducing environment variables into the context of the nginx and php-fpm servers. Use this carefully, because custom entrypoints can very easily break things.
```